src/sinol_make/commands/export/__init__.py

```diff
@@ -0,0 +1,287 @@
+                       f'                     # ze nie ma to nic wspolnego z wynikiem oitimetool dla Twojego programu)\n'
+                       f'MEMLIMIT=1000000   # limit wykorzysztania pamieci przez program oraz oitimetool (KiB)\n'
+                       f'SRCLIMIT=100     # limit na rozmiar pliku zrodlowego (KiB)\n'
+                       f'EXELIMIT=10240   # limit na rozmiar pliku wykonywalnego (KiB) = 10MB\n')
+
+    def build_debian_package(self, package_path, package_dir_name):
+        """
+        Builds debian package.
+        :param package_path: Path to directory with package.
+        :param package_dir_name: Name of directory with package.
+        :return: Path to .deb file.
+        """
+        deb_path = os.path.join(package_path, package_dir_name)
+        os.chmod(os.path.join(deb_path, 'DEBIAN'), 0o755)
+        for root, dirs, files in os.walk(os.path.join(deb_path, 'DEBIAN')):
+            for d in dirs:
+                os.chmod(os.path.join(root, d), 0o755)
+            for f in files:
+                os.chmod(os.path.join(root, f), 0o755)
+                util.fix_line_endings(os.path.join(root, f))
+
+        exit_code = os.system(f'cd "{package_path}" && dpkg-deb -v -Zxz --build {package_dir_name}')
+        if exit_code != 0:
+            util.exit_with_error("Failed to create debian package.")
+
+        return os.path.join(package_path, f'{package_dir_name}.deb')
+
+    def ocen_make_deb_package(self, tmpdir, ocen_dir):
+        """
+        Creates ocen debian package.
+        :param tmpdir: Temporary directory path.
+        :param ocen_dir: Ocen directory path.
+        :return: Path to debian package.
+        """
+        deb_template = os.path.abspath(os.path.join(util.get_templates_dir(), 'oi-ocen'))
+        deb_dir = os.path.join(tmpdir, f'oi-ocen-{self.task_id}')
+        shutil.copytree(deb_template, deb_dir)
+
+        print("Creating archive with ocen...")
+        with tarfile.open(os.path.join(tmpdir, 'ocen-linux.tgz'), 'w:gz') as tar:
+            tar.add(ocen_dir, arcname=os.path.basename(ocen_dir))
+
+        print("Creating debian package...")
+        os.makedirs(os.path.join(deb_dir, 'usr', 'share', f'oi-ocen-{self.task_id}'), exist_ok=True)
+        shutil.copyfile(os.path.join(tmpdir, 'ocen-linux.tgz'),
+                        os.path.join(deb_dir, 'usr', 'share', f'oi-ocen-{self.task_id}', 'ocen-linux.tgz'))
+
+        def replace_text(file, old, new):
+            with open(file, 'r') as f:
+                content = f.read().replace(old, new)
+            with open(file, 'w') as f:
+                f.write(content)
+
+        replace_text(os.path.join(deb_dir, 'DEBIAN', 'control'), 'TASK_ID', self.task_id)
+        replace_text(os.path.join(deb_dir, 'DEBIAN', 'postinst'), 'task_id = None', f'task_id = "{self.task_id}"')
+
+        return self.build_debian_package(tmpdir, f'oi-ocen-{self.task_id}')
```

Comment:
I have absolutely no idea what is going on here. I'll re-read this code after getting to understand the debian packages.

Reply:
All Debian packages have a “debian” directory with various package information and installation scripts inside, and everything outside this folder will be placed into root of the file system in the same directories as they are in the package. For example a file which is in usr/share/file in the package will be place into /usr/share/file in the file system.

So first the code copies the package template with necessary files in “debian” directory, creates archive with the ocen directory and places it in the usr/share/oi-ocen-{task_id}/. On installation the archive is unpacked and placed into the proper place in the home directory.